debezium-core/src/main/java/io/debezium/relational/TableIdParser.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.relational;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.debezium.text.ParsingException;
+import io.debezium.text.TokenStream;
+import io.debezium.text.TokenStream.CharacterStream;
+import io.debezium.text.TokenStream.Tokenizer;
+import io.debezium.text.TokenStream.Tokens;
+
+/**
+ * Parses identifiers into the corresponding parts of a {@link TableId}.
+ *
+ * @author Gunnar Morling.
+ */
+class TableIdParser {
+
+    private static final char SEPARATOR = '.';
+
+    public static List<String> parse(String identifier, char quotingChar) {
+        TokenStream stream = new TokenStream(identifier, new TableIdTokenizer(identifier, quotingChar), true);
+        stream.start();
+
+        List<String> parts = new ArrayList<>();
+
+        while(stream.hasNext()) {
+            parts.add(stream.consume());
+        }
+
+        return parts;
+    }
+
+    private static class TableIdTokenizer implements Tokenizer {
```

Comment:
I'm curious why you didn't want to reuse the [BasicTokenizer](https://github.com/debezium/debezium/blob/25adc3f642944cac3161f011883cb45348857bb7/debezium-core/src/main/java/io/debezium/text/TokenStream.java#L2367) that already handles multiple quoting characters, ensuring that they are paired properly. When you have a token stream for the input stream, you'd be able to parse the a name segment that is either a single- or double-quoted by doing the following:

    if (tokens.matchesAnyOf(BasicTokenizer.SINGLE_QUOTED_STRING, BasicTokenizer.DOUBLE_QUOTED_STRING)) {
        segment = tokens.consumeAnyOf(BasicTokenizer.SINGLE_QUOTED_STRING, BasicTokenizer.DOUBLE_QUOTED_STRING);
    } else {
        segment = tokens.consume();
    }

Of course, you'd want to do this for each name segment, and deal with the name segment delimiters between. But it's pretty easy.